pkg/registers/registers.go

```diff
@@ -11,6 +17,7 @@ type Register interface {
 	BitSize() uint8
 	Fields() []Field
 	Address() uint64
+	Value() interface{}
```

Comment:
why?

If someone wants a specific value of specific register, it can call

reg = FindRegisterX(registers)
reg.Raw() - which has specific type

For generic purpose there are Marshal/Unmarshal methods.

I'm ok to implement something better than what we have right now in marshal/unmarshal methods.

Having a switch of RawRegister in Marshal is not hardcoding in my view as it is done once.

Having an extra public method for that single purpose makes little sense to me. As every customer will have to guess what is that interface{} (same as with RawRegister)

Reply:
I kinda just push this into debt to be solved when we will start using generics. We'll be able to define a generic set of methods for a generic register, and it would avoid hardcoding these for each register.

I agree, it is arguable, but what do you think?